src/pybind/mgr/dashboard_v2/tests/test_block_pool.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import
+
+from cherrypy.lib.sessions import RamSession
+
+import mock
+import rbd
+
+from .helper import ControllerTestCase
+from ..controllers.auth import Auth
+from ..tools import Session
+
+
+class BlockPoolTest(ControllerTestCase):
+
+    @classmethod
+    def setup_test(cls):
+        cls._mgr_module.set_config('username', 'admin')
+        cls._mgr_module.set_config('password', Auth.password_hash('admin'))
+
+    @mock.patch('rbd.RBD.list')
+    @mock.patch('rbd.Image.stat')
+    @mock.patch('rbd.Image.features')
+    @mock.patch('rbd.Image.parent_info')
+    def test_rbd_pool_data(self, rbd_image_parent_info_mock, rbd_image_features_mock,
+                           rbd_image_stat_mock, rbd_list_mock):
+        sess_mock = RamSession()
+        with mock.patch('cherrypy.session', sess_mock, create=True):
+            self._post("/api/auth", {'username': 'admin', 'password': 'admin'})
+            self.assertStatus('201 Created')
+            self.assertJsonBody({"username": "admin"})
+            self.assertEqual(sess_mock.get(Session.USERNAME), 'admin')
+
+            rbd_list_mock.return_value = ['img1', 'img2']
+            rbd_image_stat_mock.return_value = {}
+            rbd_image_features_mock.return_value = 61
+            rbd_image_parent_info_mock.side_effect = rbd.ImageNotFound()
+            self._get('/api/block_pool/rbd_pool_data/test-pool')
+            self.assertJsonBody({
+                'status': 0,
+                'value': [
+                    {
+                        'name': 'img2',
+                        'features': 61,
+                        'features_name': 'striping, object-map, fast-diff, deep-flatten, '
```

Comment:
It's not possible to know the order of list of features names due to the way is implemented in https://github.com/openattic/ceph/pull/74/files#diff-ef9998844520eb830336059ff5907003R81

So, either you change the test code to check the result in a different way, or you change the implementation of https://github.com/openattic/ceph/pull/74/files#diff-ef9998844520eb830336059ff5907003R81 to be deterministic

Reply:
Good point!
Now the backend will always return a sorted list of features, which improves usability, thanks.